src/main/java/com/linecorp/armeria/common/metrics/DropwizardMetricConsumer.java

```diff
@@ -33,8 +33,16 @@ public DropwizardMetricConsumer(MetricRegistry metricRegistry, String metricName
     }
 
     @Override
+    public void invocationStart(Scheme scheme, String hostname, String path, Optional<String> method) {
+        final String metricName = MetricRegistry.name(metricNamePrefix, method.orElse("__unknown__"));
+        final DropwizardRequestMetrics metrics = getRequestMetrics(metricName);
+        metrics.markStart();
+    }
+
+    @Override
     public void invocationComplete(Scheme scheme, int code, long processTimeNanos, int requestSize,
```

Comment:
Could you define what 'client-side' and 'server-side' failures are? IIUC, there's only server-side failure in service metrics and there are both client-side and server-side failure in client metrics because a client invocation can fail even before hitting the wire (e.g. circuit broken or parameter validation failure). Is this what you meant?

.. and +1 for timeouts (`TimeoutException`).